--- a/NodeGraphQt/qgraphics/node_base.py
+++ b/NodeGraphQt/qgraphics/node_base.py
@@ -233,7 +233,8 @@
         border_rect = QtCore.QRectF(rect.left(), rect.top(),
                                     rect.width(), rect.height())
 
         pen = QtGui.QPen(border_color, border_width)
-        pen.setCosmetic(self.viewer().get_zoom() < 0.0)
+        viewer = self.viewer()
+        pen.setCosmetic(viewer.get_zoom() < 0.0 if viewer else False)
         path = QtGui.QPainterPath()
         path.addRoundedRect(border_rect, radius, radius)
         painter.setBrush(QtCore.Qt.NoBrush)
@@ -311,7 +312,8 @@
         border_rect = QtCore.QRectF(rect.left(), rect.top(),
                                     rect.width(), rect.height())
 
         pen = QtGui.QPen(border_color, border_width)
-        pen.setCosmetic(self.viewer().get_zoom() < 0.0)
+        viewer = self.viewer()
+        pen.setCosmetic(viewer.get_zoom() < 0.0 if viewer else False)
         painter.setBrush(QtCore.Qt.NoBrush)
         painter.setPen(pen)
         painter.drawRoundedRect(border_rect, radius, radius)